packages/sdk/src/Confidence.ts

```diff
@@ -37,15 +39,14 @@ export class Confidence implements EventSender {
     return this.config.environment;
   }
 
-  sendEvent(name: string, message?: Value.Struct) {
-    this.config.eventSenderEngine.send(this.getContext(), name, message);
+  async sendEvent(name: string, message?: Value.Struct) {
```

Comment:
q: how does this work from the user perspective in js? now we expect the api to be async, does it mean users need to make all the functions that requires to send events async?

Reply:
I don't think there's gonna be any difference from the user perspective. I think it's possible to do:
```
sendEvent(xx).catch(error => {
  console.error('Error in sendEvent:', error);
});
```

Reply:
Yes. In JS an async method always returns a promise, and that's nothing special, BUT you need to be in an async context to await a promise.